src/components/Tool/TntConnect/TntConnect.tsx

```diff
@@ -0,0 +1,392 @@
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  Divider,
+  FormControl,
+  FormControlLabel,
+  Grid,
+  LinearProgress,
+  Link,
+  Radio,
+  RadioGroup,
+  Typography,
+} from '@mui/material';
+import { styled } from '@mui/material/styles';
+import { Formik } from 'formik';
+import { useSession } from 'next-auth/react';
+import { useSnackbar } from 'notistack';
+import { useTranslation } from 'react-i18next';
+import { makeStyles } from 'tss-react/mui';
+import * as yup from 'yup';
+import { useGetContactTagListQuery } from 'src/components/Contacts/ContactDetails/ContactDetailsTab/Tags/ContactTags.generated';
+import { LoadingSpinner } from 'src/components/Settings/Organization/LoadingSpinner';
+import { ContactTagInput } from 'src/components/Tags/Tags';
+import Modal from 'src/components/common/Modal/Modal';
+import useGetAppSettings from 'src/hooks/useGetAppSettings';
+import theme from 'src/theme';
+import { uploadTnt, validateTnt } from './uploadTntConnect';
+
+const BoldTypography = styled(Typography)(() => ({
+  fontWeight: 'bold',
+}));
+
+const ContainerBox = styled(Grid)(({ theme }) => ({
+  padding: theme.spacing(3),
+  marginTop: theme.spacing(1),
+  width: '70%',
+  display: 'flex',
+  [theme.breakpoints.down('lg')]: {
+    width: '100%',
+  },
+}));
+
+const BottomBox = styled(Box)(({ theme }) => ({
+  backgroundColor: theme.palette.cruGrayLight.main,
+  width: '100%',
+  padding: theme.spacing(1),
+  display: 'flex',
+  justifyContent: 'end',
+  [theme.breakpoints.down('sm')]: {
+    paddingTop: theme.spacing(2),
+  },
+}));
+
+const OuterBox = styled(Box)(({ theme }) => ({
+  display: 'flex',
+  flexDirection: 'row',
+  justifyContent: 'center',
+  width: '100%',
+  padding: theme.spacing(3),
+}));
+
+const BorderBox = styled(Box)(({ theme }) => ({
+  marginTop: theme.spacing(2),
+  marginBottom: theme.spacing(2),
+  border: `1px solid ${theme.palette.cruGrayMedium.main}`,
+  width: '100%',
+}));
+
+const Section = styled(Box)(({ theme }) => ({
+  marginTop: theme.spacing(2),
+}));
+
+const VisuallyHiddenInput = styled('input')({
+  clip: 'rect(0 0 0 0)',
+  clipPath: 'inset(50%)',
+  height: 1,
+  overflow: 'hidden',
+  position: 'absolute',
+  bottom: 0,
+  left: 0,
+  whiteSpace: 'nowrap',
+  width: 1,
+});
+
+const useStyles = makeStyles()(() => ({
+  bulletList: {
+    margin: '0 0 10px 15px',
+    '> li': {
+      margin: '5px',
+    },
+  },
+  divider: {
+    margin: theme.spacing(2, 0),
+  },
+}));
+
+const tntSchema = yup.object({
+  selectedTags: yup.array().of(yup.string()).default([]),
+  override: yup.string().required(),
+});
+
+type Attributes = yup.InferType<typeof tntSchema>;
+
+interface Props {
+  accountListId: string;
+}
+
+const TntConnect: React.FC<Props> = ({ accountListId }: Props) => {
+  const { classes } = useStyles();
+  const { appName } = useGetAppSettings();
+  const { t } = useTranslation();
+  const { enqueueSnackbar } = useSnackbar();
+  const { data: sessionData } = useSession();
```

Comment:
Since you're using the backend, you don't need to call useSession on the frontend to pass the apiToken to the NextJs backend as ou can get it there.